Ubuntu: https://bugs.launchpad.net/ubuntu/+source/madfuload/+bug/330573
Description: Fixes selection problems with udev that prevents the firmware
loading properly
--- madfuload.orig/42-madfuload.rules.in
+++ madfuload/42-madfuload.rules.in
@@ -1,17 +1,17 @@
 # madfuload.rules - udev rules for loading firmware into M-Audio DFU devices
-
-# DEVPATH=="/*.0" selects interface 0 only
-# (some udev versions don't work with SYSFS{bInterfaceNumber})
+# Altered for Ubuntu udev system
+# Uses DEVTYPE and DEVNAME to ensure that the right device is selected
+# and the correct device file is opened
 
 # Audiophile
-ACTION=="add", SUBSYSTEM=="usb", DEVPATH=="/*.0", ENV{PRODUCT}=="763/2803/*", RUN+="@sbindir@/madfuload -l -n -f @firmwaredir@/ma003101.bin"
+ACTION=="add", SUBSYSTEM=="usb", ENV{DEVTYPE}=="usb_device", ATTRS{idVendor}=="0763", ATTRS{idProduct}=="2803", RUN+="@sbindir@/madfuload -l -n -f @firmwaredir@/ma003101.bin -D $env{DEVNAME}"
 # MobilePre
-ACTION=="add", SUBSYSTEM=="usb", DEVPATH=="/*.0", ENV{PRODUCT}=="763/2804/*", RUN+="@sbindir@/madfuload -l -3 -f @firmwaredir@/ma004103.bin"
+ACTION=="add", SUBSYSTEM=="usb", ENV{DEVTYPE}=="usb_device", ATTRS{idVendor}=="0763", ATTRS{idProduct}=="2804", RUN+="@sbindir@/madfuload -l -3 -f @firmwaredir@/ma004103.bin -D $env{DEVNAME}"
 # Sonica
-ACTION=="add", SUBSYSTEM=="usb", DEVPATH=="/*.0", ENV{PRODUCT}=="763/2805/*", RUN+="@sbindir@/madfuload -l -n -f @firmwaredir@/ma005101.bin"
+ACTION=="add", SUBSYSTEM=="usb", ENV{DEVTYPE}=="usb_device", ATTRS{idVendor}=="0763", ATTRS{idProduct}=="2805", RUN+="@sbindir@/madfuload -l -n -f @firmwaredir@/ma005101.bin -D $env{DEVNAME}"
 # Transit
-ACTION=="add", SUBSYSTEM=="usb", DEVPATH=="/*.0", ENV{PRODUCT}=="763/2806/*", RUN+="@sbindir@/madfuload -l -3 -f @firmwaredir@/ma006100.bin"
+ACTION=="add", SUBSYSTEM=="usb", ENV{DEVTYPE}=="usb_device", ATTRS{idVendor}=="0763", ATTRS{idProduct}=="2806", RUN+="@sbindir@/madfuload -l -3 -f @firmwaredir@/ma006100.bin -D $env{DEVNAME}"
 # Ozone
-ACTION=="add", SUBSYSTEM=="usb", DEVPATH=="/*.0", ENV{PRODUCT}=="763/2808/*", RUN+="@sbindir@/madfuload -l -3 -f @firmwaredir@/ma008100.bin"
+ACTION=="add", SUBSYSTEM=="usb", ENV{DEVTYPE}=="usb_device", ATTRS{idVendor}=="0763", ATTRS{idProduct}=="2808", RUN+="@sbindir@/madfuload -l -3 -f @firmwaredir@/ma008100.bin -D $env{DEVNAME}"
 
 # vim: ft=conf
